tests/repository_simulator.py

```diff
@@ -248,6 +260,8 @@ def _fetch_metadata(
 
         If version is None, non-versioned metadata is being requested.
         """
+        self.fetch_tracker.metadata.append((role, version))
```

Comment:
We append to the fetch_tracker.metadata all fetch requests. Should we append-only successful ones?